BLSQ/iaso
# WC2-116 / WC2-121 / WC2-125:  EDIT/COPY/CREATE/PUBLISH workflows versions

Thanks to @mathvdh  we have a new api to PATCH a workflow and publish it.
Front-end for edit / copy / delete and publish actions as been implemented, see [Whimsical](https://whimsical.com/entity-workflow-web-interface-5JjLtfS6my7DwLqqoaaTTT) as reference.

## Self proof reading checklist


## Changes

- new PATCH option to allow publish a workflow version
- new DELETE option to allow to soft delete a workflow version
- adapt CREATE to use reference form form entity type and specify the name 
- python tests for those
- new hooks to POST, PATCH, DELETE workflows versions
- dialog for creation and enter the name
- new actions buttons on versions list (publish, delete, copy)
- form on detail page to allow edition on name only on DRAFT versions, same for publish button


## How to test

Go to beneficiary types and click on the workflow icon in the action column.
- you should be able to add a new version as draft and choose the name
- you should be able to delete it (soft)
- you should be able to edit it (only update the name)
- you should be able to publish it on list and detail page
- you should be able to un-publish it while published
- you should be able to copy a version from published or un-published one's

## Print screen / video

<img width="1640" alt="Screenshot 2022-12-19 at 12 03 59" src="https://user-images.githubusercontent.com/12494624/208412889-b51e16e8-e78a-450a-b8bd-a76a69722810.png">
<img width="554" alt="Screenshot 2022-12-19 at 12 04 05" src="https://user-images.githubusercontent.com/12494624/208412911-569ae4a5-2ccb-4347-be58-0acb02bad511.png">


## Notes

While testing keep in mind that:
- you can change the name of a version only when the status === 'DRAFT'
- publish should set the published one in 'UNPUBLISHED' status
- you can only delete a version if the status === 'DRAFT'
- you can only un-publish a version if the status === 'PUBLISHED'
- you can only copy version where status === 'PUBLISHED' || 'UNPUBLISHED'

Should we add a confirmation dialog while publishing a workflow version ?
